### PR TITLE
154434208 improve operator search

### PR DESCRIPTION
**Pull request by rckopteri**
# Fixed
- Changed search service page operator filter a bit.

### CODE REVIEW

File: ote/src/cljs/ote/app/controller/service_search.cljs [hunk 1]
```diff
@@ -71,6 +71,10 @@
                       operator))))
 
 
+(defn- operator-in-list? [operator list]
+  "Return nil if operator is not in given list"
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/service_search.cljs]: Docstring should be placed before parameters

File: ote/src/cljs/ote/app/controller/service_search.cljs [hunk 2]
```diff
@@ -151,14 +155,18 @@
...
                    {:on-success (tuck/send-async! ->OperatorCompletionsResponse name)}))
-      app))
+      (assoc-in app [:service-search :filters :operators :results] [])))
 
   OperatorCompletionsResponse
   (process-event [result app]
-    (assoc-in app [:service-search :filters :operators :results] (:completions result)))
+    ;; Remove existing result from the list
+    (let [clean-response (filter
+                           #(nil? (operator-in-list? % (get-in app [:service-search :filters :operators :chip-results])))
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/service_search.cljs]: Use `not` to check for logical false (not the value `nil` specifically).

File: ote/src/cljs/ote/app/controller/service_search.cljs (hunk 1 above)
Comment (rckopteri) [reply ote/src/cljs/ote/app/controller/service_search.cljs]: Fixed

File: ote/src/cljs/ote/app/controller/service_search.cljs (hunk 2 above)
Comment (rckopteri) [reply ote/src/cljs/ote/app/controller/service_search.cljs]: changed

File: ote/src/cljs/ote/app/controller/service_search.cljs [hunk 3]
```diff
@@ -71,6 +71,14 @@
                       operator))))
 
 
+(defn- operator-in-list?
+  "Return nil if operator is not in given list"
+  [operator list]
+  (let [first-item (some #(= (::t-operator/id operator) (::t-operator/id %)) list)]
+    (if (nil? first-item)
```
Comment (K1ll3rF0x) [inline ote/src/cljs/ote/app/controller/service_search.cljs]: Redundant if. Negate nil?

**Pull request merged**